src/main/java/core/basesyntax/service/processor/ServiceProcessorOfShopImpl.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax.service.processor;
+
+import core.basesyntax.model.FruitRecord;
+import core.basesyntax.service.strategy.TypeStrategy;
+import java.util.List;
+
+public class ServiceProcessorOfShopImpl implements ServiceProcessorOfShop {
+    private TypeStrategy typeStrategy;
+
+    public ServiceProcessorOfShopImpl(TypeStrategy typeStrategy) {
+        this.typeStrategy = typeStrategy;
```

Comment:
It's recommended to follow the encapsulation principle. The field 'typeStrategy' should be private and final if it's not supposed to change after initialization.